cloud: add private link connection docs (#22192)
pingcap/docs: tidb-cloud/serverless-private-link-connection-to-alicloud-rds.md

---

# Set up a Private Link Connection to Alibaba Cloud ApsaraDB RDS for MySQL

This document describes how to connect to an Alibaba Cloud ApsaraDB RDS for MySQL instance using an Alibaba Cloud Endpoint Service private link connection.

## Prerequisites

1. Ensure that you have an ApsaraDB RDS for MySQL instance or have the permissions to create one.

2. Ensure that you have the following authorization to set up a load balancer and endpoint service in your own Alibaba Cloud account.

    - Manage load balancer
    - Manage endpoint services

3. Ensure that your {{{ .essential }}} in Alibaba Cloud and get its account ID and available zones. Save the information for later use. 

To view the the Alibaba Cloud account ID and available zones, do the following:

1. In the [TiDB Cloud console](https://tidbcloud.com), navigate to the cluster overview page of the TiDB cluster, and then click **Settings** > **Networking** in the left navigation pane.
2. On the **Private Link Connection For Dataflow**, click **Create Private Link Connection**.
3. You can find the Alibaba Cloud account ID and available zones.

> ## Step 1. Set up an RDS instance

Reviewer: ```suggestion
## Step 1. Set up an ApsaraDB RDS for MySQL instance
```

    - Account ID
    - Availability Zones (AZ)

To view the the Alibaba Cloud account ID and available zones, do the following:

1. In the [TiDB Cloud console](https://tidbcloud.com), navigate to the cluster overview page of the TiDB cluster, and then click **Settings** > **Networking** in the left navigation pane.
2. On the **Private Link Connection For Dataflow**, click **Create Private Link Connection**.
3. You can find the Alibaba Cloud account ID and available zones.

## Step 1. Set up an ApsaraDB RDS for MySQL instance

Identify an Alibaba Cloud ApsaraDB RDS for MySQL you want to use, or [set up a new RDS](https://www.alibabacloud.com/help/en/rds/apsaradb-rds-for-mysql/step-1-create-an-apsaradb-rds-for-mysql-instance-and-configure-databases).

To ensure successful connectivity, your ApsaraDB RDS for MySQL instance must meet the following requirements:

- Region match: the instance must reside in the same Alibaba Cloud region as your {{{ .essential }}} cluster.
- AZ (Availability Zone) availability: the availability zones must overlap with those of your {{{ .essential }}} cluster.

> - Network accessibility: the instance must be accessible within the VPC, with an appropriately configured IP allowlist.

Reviewer 2: 这啥意思？

> - Network accessibility: the instance must be accessible within the VPC, with an appropriately configured IP allowlist.

Author: use whitelist rather than allowlist
<img width="2990" height="718" alt="image" src="https://github.com/user-attachments/assets/07486321-7078-4350-8883-eccfdd7bc243" />

---
title: Connect to Alibaba Cloud ApsaraDB RDS for MySQL via a Private Link Connection
summary: Learn how to connect to an Alibaba Cloud ApsaraDB RDS for MySQL instance using an Alibaba Cloud Endpoint Service private link connection.
---

# Connect to Alibaba Cloud ApsaraDB RDS for MySQL via a Private Link Connection 

> This document describes how to connect a {{{ .essential }}} cluster to an [Alibaba Cloud ApsaraDB RDS for MySQL](https://www.alibabacloud.com/en/product/apsaradb-for-rds-mysql) instance using an Alibaba Cloud Endpoint Service private link connection.

Reviewer 3: ```suggestion
This document describes how to connect a {{{ .essential }}} cluster to an [Alibaba Cloud ApsaraDB RDS for MySQL](https://www.alibabacloud.com/en/product/apsaradb-for-rds-mysql) instance using an [Alibaba Cloud Endpoint Service private link connection](/tidb-cloud/serverless-private-link-connection.md).
```

## Prerequisites

- You have an existing ApsaraDB RDS for MySQL instance or the permissions required to create one.

- Verify that your account has the following permissions to manage networking components:

    - Manage load balancer
    - Manage endpoint services

- Your {{{ .essential }}} cluster is on Alibaba Cloud, and it is active. Retrieve and save the following details for later use:

>     - Account ID

Reviewer 3: ```suggestion
    - Alibaba Cloud account ID
```

    - Availability Zones (AZ)

To view the the Alibaba Cloud account ID and availability zones, do the following:

1. In the [TiDB Cloud console](https://tidbcloud.com), navigate to the cluster overview page of the TiDB cluster, and then click **Settings** > **Networking** in the left navigation pane.
2. In the **Private Link Connection For Dataflow** area, click **Create Private Link Connection**.
3. In the displayed dialog, you can find the Alibaba Cloud account ID and availability zones.

## Step 1. Set up an ApsaraDB RDS for MySQL instance

Identify an Alibaba Cloud ApsaraDB RDS for MySQL that you want to use, or [create a new RDS](https://www.alibabacloud.com/help/en/rds/apsaradb-rds-for-mysql/step-1-create-an-apsaradb-rds-for-mysql-instance-and-configure-databases).

Your ApsaraDB RDS for MySQL instance must meet the following requirements:

- Region match: the instance must reside in the same Alibaba Cloud region as your {{{ .essential }}} cluster.
- AZ (Availability Zone) availability: the availability zones must overlap with those of your {{{ .essential }}} cluster.

> - Network accessibility: the instance must be configured with proper IP whitelist and be accessible within the VPC.

Reviewer 3: ```suggestion
- Network accessibility: the instance must be configured with a proper IP allowlist and be accessible within the VPC.
```

> **Note**
>
> Cross-region connections for ApsaraDB RDS for MySQL are not supported.

## Step 2. Expose the ApsaraDB RDS for MySQL instance as an endpoint service

You need to set up the load balancer and the endpoint service in the Alibaba Cloud console.

### Step 2.1. Set up the load balancer

> Set up the load balancer in the same region of your ApsaraDB RDS for MySQL as follows:

Reviewer 3: ```suggestion
Set up the load balancer in the same region as your ApsaraDB RDS for MySQL as follows:
```

    - **Server Group Type**: select `IP`
    - **VPC**: enter the VPC where your ApsaraDB RDS for MySQL is located
    - **Backend Server Protocol**: select `TCP`

2. Click the created server group to add backend servers, and then add the IP address of your ApsaraDB RDS for MySQL instance. 

    You can ping the RDS endpoint to get the IP address.
 
3. Go to [NLB](https://slb.console.alibabacloud.com/nlb) to create a network load balancer. Provide the following information:

    - **Network Type**: select `Internal-facing`
    - **VPC**: select the VPC where your ApsaraDB RDS for MySQL is located
    - **Zone**: it must overlap with your {{{ .essential }}} cluster
    - **IP Version**: select `IPv4`

4. Find the load balancer you created, and then click **Create Listener**. Provide the following information:
    
    - **Listener Protocol**: select `TCP`
    - **Listener Port**: enter the database port, for example, `3306` for MySQL
    - **Server Group**: choose the server group you created in the previous step
  
### Step 2.2. Set up an endpoint service

> To set up the endpoint service in the same region of your ApsaraDB RDS for MySQL, take the following steps:

Reviewer 3: ```suggestion
To set up the endpoint service in the same region as your ApsaraDB RDS for MySQL, take the following steps:
```

1. Go to [Endpoint Service](https://vpc.console.alibabacloud.com/endpointservice) to create an endpoint service. Provide the following information:

    - **Service Resource Type**: select `NLB`
    - **Select Service Resource**: select all zones that NLB is in, and choose the NLB that you created in the previous step
    - **Automatically Accept Endpoint Connections**: it is recommended to choose `No`

2. Go to the details page of the endpoint service, and copy the **Endpoint Service Name**, for example, `com.aliyuncs.privatelink.<region>.xxxxx`. You need to use it for TiDB Cloud later. 

> 3. On the detail page of the endpoint service, click the **Service Whitelist** tab, click **Add to Whitelist**, and then enter the TiDB Cloud account ID. 

Reviewer 3: ```suggestion
3. On the details page of the endpoint service, click the **Service Whitelist** tab, click **Add to Whitelist**, and then enter the TiDB Cloud account ID. 
```
